### DISCUSSION TITLE
Valid tags at ignorecommand ?

### DISCUSSION THREAD

**Discussion by pfgomezv** [fail2ban/fail2ban#3798] [Q&A] [answered]
I've tried to make a filter/whitelist for geoip and manual defined networks, is very useful to see the user (\<F-USER\>) at each origin suspicious valid connection to alert if the accounts are compromised.   @sebres show  the best aproach to make this type of filter at [Time-based Whitelinting...](https://github.com/fail2ban/fail2ban/issues/2013) and [Code structure for filter..](https://github.com/fail2ban/fail2ban/issues/2252) responses.

At man pages in _ignorecommand_ section says:

> ...Like	ACTION FILES, tags like <ip> are can  be  included  in  the  ignorecommand  value...

I created a jail and put a _ignorecommand_ entry.
I call  a bash script and can pass <ip> tag as argument , but can't see the <F-USER> value.
My jail.local looks like:
```
[geopostfix]
enabled = true
logpath  = %(postfix_log)s
backend  = %(postfix_backend)s
filter   = geopostfix
action   = geopostfix
port     = 25,465,587
maxretry = 1
ignorecommand = /etc/fail2ban/scripts/check-ip.sh "<ip>" "<F-USER>"

```
I log the parameters passed to script and see this at fail2ban.log  (view first line):

```
...
2024-07-17 12:21:27,247 fail2ban.test                  : IGNORE  [geopostfix-check-ip]  10.9.11.27 -- Name: geopostfix -- User:<F-USER> 
2024-07-17 12:21:27,291 fail2ban.filter                : INFO    [geopostfix-check-ip]  10.9.11.27 is at group/country: "VPN" and the result policy is "ACCEPT" 
2024-07-17 12:21:27,298 fail2ban.filter                : INFO    [geopostfix-check-ip] IP: 10.9.11.27 conection ignored for this jail 
2024-07-17 12:21:27,304 fail2ban.filter         [13330]: INFO    [geopostfix] Ignore 10.9.11.27 by command
...
```

I can use \<F-USER\> tag  at  action file in ban / unban sections , but not at  ignorecommand in jail section.
I make something wrong? or missunderstood  the sintaxis?

version: 
fail2ban   0.10.2-2
Ubuntu 18.04 LTS

**Comment by sebres** [comment, marked as answer]
> I can use `<F-USER>` tag at action file in ban / unban sections , but not at `ignorecommand` in jail section.

No, normally you can do that, every action tag is available for `ignorecommand` too...
It is also covered in the test suite at several places, also for `<F-USER>` tag:
https://github.com/fail2ban/fail2ban/blob/766d2b8d742cf207f3321b27bf74fce2760a8eb7/fail2ban/tests/filtertestcase.py#L579

What you probably missing is the capturing definition in the filter - the RE matching the line must contain `<F-USER>...</F-USER>` capturing group.
Another mistake could be enclosing `'<F-USER>'` in the single quotes, but I see you used double quotes there (unless you changed it later without reload the jail, what is expected to apply all changes).

**Reply by pfgomezv** [reply]
Thank you @sebres .
I check this things and were OK, so check  the file [fail2ban/fail2ban/tests/filtertestcase.py](https://github.com/fail2ban/fail2ban/blob/766d2b8d742cf207f3321b27bf74fce2760a8eb7/fail2ban/tests/filtertestcase.py#L579) that you mention, and my file do not have these line with <F-USER> tag.
So i manual upgrade  ti version 1.1.0  as documentation says at [Manual Install/Upgrade](https://github.com/fail2ban/fail2ban/wiki/How-to-install-or-upgrade-fail2ban-manually) and work like charm!
Thanks a lot!

**Reply by sebres** [reply]
Hmm, how old was your version?.. 
This capability got implemented in 9b6d17d07e2f00dd287a7ccb5f30b95685d6f935 (6 years ago) and firstly released with 0.10.4...

**Reply by pfgomezv** [reply]
my version was stock  ubuntu 18.04. very old...

> version:
fail2ban 0.10.2-2
Ubuntu 18.04 LTS
